src/communication/xcWebSocketSubscriber.ts

```diff
@@ -132,26 +135,20 @@ export class DefaultSubscriber implements Subscriber {
     };
 
 
-    getSnapshot(componentName: string, stateMachineName: string, getSnapshotListener: (items: Array<Packet>) => void): void {
-        let replyTopic = uuid();
-        let thisSubscriber = this;
-        this.observableMsg
-            .map((rawMessage: MessageEvent) => thisSubscriber.deserializer.deserialize(rawMessage.data || rawMessage))
-            .filter((data: DeserializedData) => (data.command === Commands[Commands.snapshot] && data.topic === replyTopic))
-            .subscribe((data: DeserializedData) => {
-                thisSubscriber.sendUnsubscribeRequestToTopic(replyTopic, Kinds.Snapshot);
-                getSnapshotListener(thisSubscriber.getJsonDataFromSnapshot(data.stringData, data.topic));
-            });
-        this.sendSubscribeRequestToTopic(replyTopic, Kinds.Snapshot);
-        let dataToSendSnapshot = this.getDataToSendSnapshot(componentName, stateMachineName, replyTopic);
-        this.webSocket.send(thisSubscriber.serializer.convertToWebsocketInputFormat(dataToSendSnapshot));
-    };
-
-    getSnapshotPromise(componentName: string, stateMachineName: string): Promise<Array<Packet>> {
+    getSnapshot(componentName: string, stateMachineName: string): Promise<Array<Packet>> {
         return new Promise((resolve, reject) => {
-            this.getSnapshot(componentName, stateMachineName, (items: Array<Packet>) => {
-                resolve(items);
-            });
+            let replyTopic = uuid();
+            let thisSubscriber = this;
+            this.observableMsg
+                .map((rawMessage: MessageEvent) => thisSubscriber.deserializer.deserialize(rawMessage.data || rawMessage))
+                .filter((data: DeserializedData) => (data.command === Commands[Commands.snapshot] && data.topic === replyTopic))
+                .subscribe((data: DeserializedData) => {
```

Comment:
use first method too

Reply:
this link [https://medium.com/@benlesh/rxjs-observable-interop-with-promises-and-async-await-bebb05306875](https://medium.com/@benlesh/rxjs-observable-interop-with-promises-and-async-await-bebb05306875) can help you work with promise and observable together